model/MessageQueue.go

```diff
@@ -0,0 +1,50 @@
+package model
+
+import (
+	"container/heap"
+	"sync"
+
+	"github.com/puchkovki/distr-model/model"
+	"github.com/puchkovki/distr-model/priorityq"
+)
+
+// A MessageQueue is a PriorityQueue structure with mutex for parallel usage.
+type MessageQueue struct {
+	queue priorityq.PriorityQueue
+	mutex sync.Mutex
+}
+
+// Initialiaze establishes the heap invariants.
+func (mq *MessageQueue) Initialiaze() {
```

Comment:
It should be a constructor. Smth like that:

```
func NewMessageQueue() (mq MessageQueue) {
	heap.Init(&mq.queue)
}
```

Reply:
Fixed